### DISCUSSION TITLE
Taxes rules

### DISCUSSION THREAD

**Discussion by josias-hosang** [swellstores/community#428] [Help]
How do taxes rules exactly work (documentation would be nice)? It seems to me priority is ignored. If there are 2 rules w/o categories set, both will be applied. I tested this with same and differing priorities.

If a tax rule is configured to shipping costs but not any products or categories, shouldn't it then only be applied on shipment costs?

It would be nice to have a default tax rule but to override it for some products or categories. Otherwise one can easily forget to reconfigure tax rules after adding a category.

**Comment by coderreco** [comment]
Hey @josias-hosang,

A couple of things I know about how taxes calculate:

Priority definitely isn't ignored. I accidentally had one of my tax rules set to priority 2, and it was calculating tax in a compounding fashion.

For 'Product tax classes' and 'Customer tax classes' you can specify a class that this tax rule should be restricted to. See picture:
![CleanShot 2023-11-03 at 14 22 16@2x](https://github.com/swellstores/community/assets/73268171/1b2e4612-5229-4ecf-95d1-090f495cf23d)
In the screenshot, this tax rule will only apply to products in the 'Standard' tax class. But if I uncheck 'Standard', so neither 'Standard' nor 'Digital' are checked it will automatically apply that tax to all Product tax classes. 

To only tax shipping, here is a workaround:
1. In the developer console create a new tax class for shipping
`POST` `/settings/taxes/product_classes`
```
{
"id": "shipping",
"name": "Shipping"
}
```
2. Edit your tax rule, and select your new product tax class. Make sure no other tax classes are selected.
3. Make sure 'Apply tax to shipping' is selected.

Since no products are assigned to that product tax class, it won't apply to any. But since you are designating a specific tax class to apply that tax to it won't default to all. Now it only applies to shipping.

**Reply by josias-hosang** [reply]
I understand (also how priorities are working), found also now the helpful FAQ and help page. Thanks.

In Switzerland we have 2 tax rates, one for food & beverage and a higher one for all other products and services. So we have to assign to each product category one or the other tax rule? I can't define a standard rule, otherwise taxes are added. Again, I see a problem one forgets to add correct tax rule after adding a new product category (and no taxes are calculated). Do you see a solution to this problem?

Another missing point is, that by law each tax rate (and it's value) has to be listed in checkout. But at the moment only a grand total and containing taxes for shipping are listed. Is there a possibility this can be customized to run the shop fulfilling our law?
